packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/document_migrator.ts

```diff
@@ -68,42 +29,49 @@ interface DocumentMigratorOptions {
 }
 
 /**
- * Manages migration of individual documents.
+ * Manages transformations of individual documents.
  */
 export interface VersionedTransformer {
-  migrationVersion: SavedObjectsMigrationVersion;
-  migrate: MigrateFn;
-  migrateAndConvert: MigrateAndConvertFn;
-  prepareMigrations: () => void;
+  /**
+   * Migrates a document to its latest version.
+   */
+  migrate: (doc: SavedObjectUnsanitizedDoc) => SavedObjectUnsanitizedDoc;
+  /**
+   * Migrates a document to the latest version and applies type conversions if applicable.
+   * Also returns any additional document(s) that may have been created during the transformation process.
+   */
+  migrateAndConvert: (doc: SavedObjectUnsanitizedDoc) => SavedObjectUnsanitizedDoc[];
+  /**
+   * Converts a document down to the specified version.
+   */
+  transformDown: (
+    doc: SavedObjectUnsanitizedDoc,
+    options: { targetTypeVersion: string }
+  ) => SavedObjectUnsanitizedDoc;
```

Comment:
Note: I highly suspect that given what we're intending to use the feature for, we will need some kind of `getAtVersion` API that either convert up or down depending on the doc and target version. However it will likely just delegates to the existing APIs, and I didn't want to implement it before having a better understanding of how the SOR will use the feature in https://github.com/elastic/kibana/issues/150312